### DISCUSSION TITLE
Deploy nextjs on Azure

### DISCUSSION THREAD

**Discussion by gsambrotta** [vercel/next.js#21640] [Help]
Hi everybody, 
I have a simple Nextjs app without backend and WITH nextjs i18n, and I need to deploy it on Azure and I cannot understand how to do it. 
I try to check first locally if the process work, this is my file structure:
```
 babel.config.json
i18n.js
.next/
 next.config.js
 node_modules/
 package.json
 package-lock.json
public/
```

Steps I run are:
`npm i`
`npm build`
`npm start`very

When I run `npm start` I get:
`no such file or directory /pages`

My questions are:
- Why do I need pages? Should all the code be bundles and exist in the .next/ folder? Shouldn't I just need the .next/ folder in order to run my app?
- Do i need the node modeuls? Right now I have them because I `npm install` in my server but I was wondering If also node modules are bundled.. Is quite a lof ot code to host on the server and to rerun at each deploy...

The only valid resource I found about Nextjs and Azure is this video https://www.youtube.com/watch?v=HLsx0iraA-Y
I wonder if I need to add a custom server to make it work.
But then, shouldn't at least work in my local? 

I'm a newby to deployment so please be kind if some very basic concept are misinterpretation

**Comment by aaronpowell** [comment]
How are you handling the deployment? We recently release improved support for Next.js on Static Web Apps (see https://techcommunity.microsoft.com/t5/apps-on-azure-blog/extending-next-js-support-in-azure-static-web-apps/ba-p/3627975) that supports features like i18n, so that might make it easier to get up and running.
